File: app/src/main/java/com/mohamedtaha/imagine/ui/activities/SwipePagesActivity.java
package com.mohamedtaha.imagine.ui.activities;

import android.app.NotificationManager;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.WindowManager;
import android.widget.ProgressBar;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.mohamedtaha.imagine.Adapter.AdapterForSwipe;
import com.mohamedtaha.imagine.R;
import com.mohamedtaha.imagine.helper.HelperClass;
import com.mohamedtaha.imagine.helper.SharedPerefrenceHelper;
import com.mohamedtaha.imagine.helper.ShowDialog;
import com.mohamedtaha.imagine.mvp.model.ModelAzkar;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;
import java.util.Locale;
import butterknife.BindView;
import butterknife.ButterKnife;

import static com.mohamedtaha.imagine.helper.Images.SAVE_POSITION;
import static com.mohamedtaha.imagine.notification.morningAzkar.MorningAzkarAlarmReceiver.NOTIFICATION_ID_NUMBER_AZKAR;
import static com.mohamedtaha.imagine.notification.morningAzkar.MorningAzkarAlarmReceiver.SAVE_POSITION_MORNING_AZKAR;
import static com.mohamedtaha.imagine.notification.morningAzkar.MorningAzkarAlarmReceiver.TIME_SEND_MORNING_AZKAR;
import static com.mohamedtaha.imagine.notification.quran.AlarmReceiver.NOTIFICATION_ID;
import static com.mohamedtaha.imagine.notification.quran.AlarmReceiver.SAVE_Position_Notification;
import static com.mohamedtaha.imagine.notification.quran.AlarmReceiver.TIME_SEND;
import static com.mohamedtaha.imagine.ui.activities.NavigationDrawaberActivity.SAVE_ALL_IMAGES;
import static com.mohamedtaha.imagine.ui.activities.NavigationDrawaberActivity.SAVE_PAGE;
import static com.mohamedtaha.imagine.ui.fragments.AzkarFragment.SAVE_AZKAR;
import static com.mohamedtaha.imagine.ui.fragments.AzkarFragment.SAVE_POTION_AZKAR;
import static com.mohamedtaha.imagine.ui.fragments.SwarFragment.SAVE_IMAGES;
import static com.mohamedtaha.imagine.ui.fragments.SwarFragment.SAVE_STATE;
public class SwipePagesActivity extends AppCompatActivity {
    ArrayList<Integer> images, imagesNotification, imagesFirst;
    public static final String IS_TRUE = "is_true";
    public static final String IS_TRUE_AZKAR = "is_true_azkar";
    @BindView(R.id.SwipePagesActivity_PB)
    ProgressBar SwipePagesActivityPB;
//    @BindView(R.id.SwipePagesActivity_VP)
//    RtlViewPager SwipePagesActivityVP;
//    @Inject
//    List<ModelAzkar> modelAzkarList;
//    @Inject
//    SharedPreferences sharedPreferences;
    private int save_position;
    private int save_position_azkar;
    private int position = 1;
    private int position_azkar = 0;
    private Bundle bundle;
    int notificationId = -1;
    int notification_id_morning_azkar = -1;
    private String language_name;
    private boolean screenOn;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_swipe_pages);
        ButterKnife.bind(this);
        bundle = getIntent().getExtras();
        //to Check before change Language
        language_name = Locale.getDefault().getLanguage();
        if (!language_name.equals("ar")) {
            HelperClass.change_language("ar", this);
        }
        NotificationManager notificationManager = (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
//        SwipePagesActivityComponent swipePagesActivityComponent = DaggerSwipePagesActivityComponent.builder()
//                .sharedPreferencesModule(new SharedPreferencesModule(this)).build();
//        swipePagesActivityComponent.inject(this);
     //   screenOn = sharedPreferences.getBoolean(getString(R.string.switch_key), true);
        getScreenOn(screenOn);
        //for close Notification
        notificationId = getIntent().getIntExtra(NOTIFICATION_ID, -1);
        notification_id_morning_azkar = getIntent().getIntExtra(NOTIFICATION_ID_NUMBER_AZKAR, -1);
        int timeSend = getIntent().getIntExtra(TIME_SEND, -1);
        int time_send_morning_azkar = getIntent().getIntExtra(TIME_SEND_MORNING_AZKAR, -1);
        if (notificationId >= 0) {
            getArgemnetsNotification();
            createImageNotification();
            AdapterForSwipe adapterForSwipe = new AdapterForSwipe(this, imagesNotification);
//            SwipePagesActivityVP.setAdapter(adapterForSwipe);
//            SwipePagesActivityVP.setCurrentItem(notificationId);
            Log.d("TAG", " " + notificationId);
            if (notificationManager != null) {
                notificationManager.cancel(timeSend);
            }
        } else if (notification_id_morning_azkar >= 0) {
            getArgemnetsForNotificationAzkar();
         //   AdapterForAzkarSwipe adapterForAzkarSwipe = new AdapterForAzkarSwipe(SwipePagesActivity.this, modelAzkarList);
            if (notification_id_morning_azkar == 27) {
//                SwipePagesActivityVP.setAdapter(adapterForAzkarSwipe);
//                SwipePagesActivityVP.setCurrentItem(notification_id_morning_azkar);

            } else {
//                SwipePagesActivityVP.setAdapter(adapterForAzkarSwipe);
//                SwipePagesActivityVP.setCurrentItem(notification_id_morning_azkar);
            }
            if (notificationManager != null) {
                notificationManager.cancel(time_send_morning_azkar);
            }
        } else if (bundle.getBoolean(SAVE_STATE)) {
            getArgemnets();
            createImage();
            AdapterForSwipe adapterForSwipe = new AdapterForSwipe(this, images, postiopn ->
                    ShowDialog.showDialog(SwipePagesActivity.this, save_position, getString(R.string.save_position)));
//
//            SwipePagesActivityVP.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
//                @Override
//                public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
//                }
//
//                @Override
//                public void onPageSelected(int position) {
//                    save_position = position;
//                }
//
//                @Override
//                public void onPageScrollStateChanged(int state) {
//                }
//            });
//            SwipePagesActivityVP.setAdapter(adapterForSwipe);
//            if (SharedPerefrenceHelper.getBoolean(this, IS_TRUE, false)) {
//                ShowDialog.showDialogForRetrieveReadingSora(this, SwipePagesActivityVP, position);
//            } else {
//                SwipePagesActivityVP.setCurrentItem(position);
//            }
        } else if (bundle.getBoolean(SAVE_PAGE)) {
            getImagesFirst();
            createImagesFirst();
            AdapterForSwipe adapterForSwipe = new AdapterForSwipe(this, imagesFirst, positon ->
                    ShowDialog.showDialog(SwipePagesActivity.this, save_position, getString(R.string.save_position)));
//            SwipePagesActivityVP.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
//                @Override
//                public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
//                }
//
//                @Override
//                public void onPageSelected(int position) {
//                    save_position = position;
//                }
//
//                @Override
//                public void onPageScrollStateChanged(int state) {
//                }
//            });
//            SwipePagesActivityVP.setAdapter(adapterForSwipe);
//            int position = SharedPerefrenceHelper.getInt(getApplicationContext(), SAVE_IMAGES, 0);
//            SwipePagesActivityVP.setCurrentItem(position);
        } else {
            getArgemnetsForAzkar();
          //  AdapterForAzkarSwipe adapterForAzkarSwipe = new AdapterForAzkarSwipe(SwipePagesActivity.this, modelAzkarList, modelAzkar ->
          //          ShowDialog.showDialogAzkar(SwipePagesActivity.this, save_position_azkar, getString(R.string.save_position_alzekr), modelAzkar));
//            SwipePagesActivityVP.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
//                @Override
//                public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
//                }
//
//                @Override
//                public void onPageSelected(int position) {
//                    save_position_azkar = position;
//                }
//
//                @Override
//                public void onPageScrollStateChanged(int state) {
//                }
//            });
//            SwipePagesActivityVP.setAdapter(adapterForAzkarSwipe);
//            if (SharedPerefrenceHelper.getBooleanForAzkar(this, IS_TRUE_AZKAR, false)) {
//                ShowDialog.showDialogForRetrieveAzkar(this, SwipePagesActivityVP, position_azkar);
//            } else {
//                SwipePagesActivityVP.setCurrentItem(position_azkar);
//            }
        }
    }

    private void getScreenOn(boolean isScreenOn) {
        if (isScreenOn) {
            getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
        } else {
            getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
        }
    }

    private void getArgemnets() {
        if (bundle != null) {
            images = bundle.getIntegerArrayList(SAVE_IMAGES);
            position = bundle.getInt(SAVE_POSITION);
        }
    }

    private void createImage() {
        for (int i = 0; i < images.size(); i++) {
            bundle.putInt(SAVE_IMAGES, images.get(i));
            bundle.putInt(SAVE_POSITION, position);
        }
        SwipePagesActivityPB.setVisibility(View.GONE);
    }

    private void getArgemnetsNotification() {
        if (bundle != null) {
            imagesNotification = bundle.getIntegerArrayList(SAVE_Position_Notification);
            Log.d("TAG", " " + imagesNotification);
        }
    }

    private void createImageNotification() {
        if (imagesNotification != null) {
            for (int i = 0; i < imagesNotification.size(); i++) {
                bundle.putInt(SAVE_Position_Notification, imagesNotification.get(i));
            }
            Log.d("TAG", " " + imagesNotification);
            SwipePagesActivityPB.setVisibility(View.GONE);
        }
    }

    private void getAzkarNotification() {
        if (bundle != null) {
            imagesNotification = bundle.getIntegerArrayList(SAVE_Position_Notification);
            Log.d("TAG", " " + imagesNotification);
        }
    }

    private void createAzkarNotification() {
        if (imagesNotification != null) {
            for (int i = 0; i < imagesNotification.size(); i++) {
                bundle.putInt(SAVE_Position_Notification, imagesNotification.get(i));
            }
            Log.d("TAG", " " + imagesNotification);
            SwipePagesActivityPB.setVisibility(View.GONE);
        }
    }

    private void getImagesFirst() {
        if (bundle != null) {
            imagesFirst = bundle.getIntegerArrayList(SAVE_ALL_IMAGES);
        }
    }

    private void createImagesFirst() {
        if (imagesFirst != null) {
            for (int i = 0; i < imagesFirst.size(); i++) {
                bundle.putInt(SAVE_ALL_IMAGES, imagesFirst.get(i));
            }
            SwipePagesActivityPB.setVisibility(View.GONE);
        }
    }

    private void getArgemnetsForAzkar() {
        if (bundle != null) {
            Type listType = new TypeToken<List<ModelAzkar>>() {
            }.getType();
            String st = bundle.getString(SAVE_AZKAR);
          //  modelAzkarList = new Gson().fromJson(st, listType);
            position_azkar = bundle.getInt(SAVE_POTION_AZKAR);
        }
        SwipePagesActivityPB.setVisibility(View.GONE);
    }

    private void getArgemnetsForNotificationAzkar() {
        if (bundle != null) {
            Type listType = new TypeToken<List<ModelAzkar>>() {
            }.getType();
            String st = bundle.getString(SAVE_POSITION_MORNING_AZKAR);
            //modelAzkarList = new Gson().fromJson(st, listType);
            position_azkar = bundle.getInt(NOTIFICATION_ID_NUMBER_AZKAR);
        }
        SwipePagesActivityPB.setVisibility(View.GONE);
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        SharedPerefrenceHelper.putInt(this, SAVE_IMAGES, save_position);
        SharedPerefrenceHelper.putInt(this, SAVE_AZKAR, save_position_azkar);
        super.onSaveInstanceState(outState);
    }

    @Override
    protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        save_position = savedInstanceState.getInt(SAVE_IMAGES);
        save_position_azkar = savedInstanceState.getInt(SAVE_AZKAR);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        overridePendingTransition(R.anim.item_anim_no_thing, R.anim.item_anim_slide_from_bottom);
    }

}

File: app/src/main/java/com/mohamedtaha/imagine/ui/activities/SwipePagesActivity.kt
package com.mohamedtaha.imagine.ui.activities

import android.app.NotificationManager
import android.content.DialogInterface
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.WindowManager
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.asLiveData
import androidx.viewpager.widget.ViewPager
import butterknife.OnClick
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.mohamedtaha.imagine.R
import com.mohamedtaha.imagine.adapter.AdapterForAzkarSwipe
import com.mohamedtaha.imagine.adapter.AdapterForSwipe
import com.mohamedtaha.imagine.databinding.ActivitySwipePagesBinding
import com.mohamedtaha.imagine.datastore.DataStoreViewModel
import com.mohamedtaha.imagine.helper.HelperClass
import com.mohamedtaha.imagine.helper.SharedPerefrenceHelper
import com.mohamedtaha.imagine.helper.ShowDialog
import com.mohamedtaha.imagine.helper.images
import com.mohamedtaha.imagine.helper.images.SAVE_POSITION
import com.mohamedtaha.imagine.mvp.model.ModelAzkar
import com.mohamedtaha.imagine.notification.morningAzkar.MorningAzkarAlarmReceiver
import com.mohamedtaha.imagine.notification.quran.AlarmReceiver
import com.mohamedtaha.imagine.ui.home.activity.NavigationDrawaberActivity
import com.mohamedtaha.imagine.ui.home.fragment.AzkarFragment
import com.mohamedtaha.imagine.ui.home.fragment.SwarFragment
import com.mohamedtaha.imagine.ui.home.viewModel.SwarAndPartsViewModel
import com.mohamedtaha.imagine.util.ClickListener
import com.mohamedtaha.imagine.util.DialogUtils
import dagger.hilt.android.AndroidEntryPoint
import java.util.*
import javax.inject.Inject

@AndroidEntryPoint
class SwipePagesActivity : AppCompatActivity() {
    private lateinit var binding: ActivitySwipePagesBinding
    private val viewModel: SwarAndPartsViewModel by viewModels()
    private val dataStoreViewModel by viewModels<DataStoreViewModel>()
    var images: List<Int>? = null
    var imagesNotification: ArrayList<Int>? = null
    var imagesFirst: ArrayList<Int>? = null
    private lateinit var modelAzkarList: List<ModelAzkar>

    private var save_position = 0
    private var save_position_azkar = 0
    private var position = 1
    private var position_azkar = 0

    @Inject
    lateinit var bundle: Bundle
    var notificationId = -1
    var notificationIdMorningAzkar = -1
    private var language_name: String? = null
    private val screenOn = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySwipePagesBinding.inflate(layoutInflater)
        setContentView(binding.root)
        viewModel.addImagesList()



        bundle = intent?.extras!!
        //to Check before change Language
        language_name = Locale.getDefault().language
        if (language_name != "ar") {
            HelperClass.change_language("ar", this)
        }
        val notificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
        getScreenOn(screenOn)
        //for close Notification
        notificationId = intent.getIntExtra(AlarmReceiver.NOTIFICATION_ID, -1)
        notificationIdMorningAzkar =
            intent.getIntExtra(MorningAzkarAlarmReceiver.NOTIFICATION_ID_NUMBER_AZKAR, -1)
        val timeSend = intent.getIntExtra(AlarmReceiver.TIME_SEND, -1)
        val timeSendMorningAzkar =
            intent.getIntExtra(MorningAzkarAlarmReceiver.TIME_SEND_MORNING_AZKAR, -1)
        if (notificationId >= 0) {
            argemnetsNotification
            createImageNotification()
         //   val adapterForSwipe = AdapterForSwipe(imagesNotification)
          //  binding.SwipePagesActivityVP.setAdapter(adapterForSwipe);
          //  binding.SwipePagesActivityVP.setCurrentItem(notificationId);
            Log.d("TAG", " $notificationId")
            notificationManager.cancel(timeSend)
        } else if (notificationIdMorningAzkar >= 0) {
            argemnetsForNotificationAzkar
            val adapterForAzkarSwipe = AdapterForAzkarSwipe(this, modelAzkarList);
            if (notificationIdMorningAzkar == 27) {
                binding.SwipePagesActivityVP.setAdapter(adapterForAzkarSwipe);
                binding.SwipePagesActivityVP.setCurrentItem(notificationIdMorningAzkar);
            } else {
                binding.SwipePagesActivityVP.setAdapter(adapterForAzkarSwipe);
                binding.SwipePagesActivityVP.setCurrentItem(notificationIdMorningAzkar);
            }
            notificationManager.cancel(timeSendMorningAzkar)
        } else if (bundle.getBoolean(SwarFragment.SAVE_STATE)) {
            position = bundle.getInt(SAVE_POSITION)
            viewModel.allImages.observe(this) {
                images = it
                createImage()
                declareAdapter()
                observeViewPager()
                observeSaveData()

            }
        } else if (bundle.getBoolean(NavigationDrawaberActivity.SAVE_PAGE)) {
            getImagesFirst()
            createImagesFirst()
            val adapterForSwipe = AdapterForSwipe(
                imagesFirst!!,object: ClickListener<Int>{
                    override fun onClick(view: View?, position: Int) {
                        Toast.makeText(this@SwipePagesActivity,"Click",Toast.LENGTH_LONG).show()
                    }

                }
            )
//            { positon: Int ->
//                ShowDialog.showDialog(
//                    this@SwipePagesActivity,
//                    save_position,
//                    getString(R.string.save_position)
//                )
//            }
            //            SwipePagesActivityVP.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
//                @Override
//                public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
//                }
//
//                @Override
//                public void onPageSelected(int position) {
//                    save_position = position;
//                }
//
//                @Override
//                public void onPageScrollStateChanged(int state) {
//                }
//            });
//            SwipePagesActivityVP.setAdapter(adapterForSwipe);
//            int position = SharedPerefrenceHelper.getInt(getApplicationContext(), SAVE_IMAGES, 0);
//            SwipePagesActivityVP.setCurrentItem(position);
        } else {
            getAzkarList()
            val adapterForAzkarSwipe = AdapterForAzkarSwipe(
                this,
                modelAzkarList,
                object : AdapterForAzkarSwipe.showDetail {
                    override fun showDetails(modelAzkar: ModelAzkar?) {
                        // ShowDialog.showDialogAzkar(reques, save_position_azkar, getString(R.string.save_position_alzekr), modelAzkar)
                    }

                })

            binding.SwipePagesActivityVP.addOnPageChangeListener(object :
                ViewPager.OnPageChangeListener {
                override fun onPageScrolled(
                    position: Int,
                    positionOffset: Float,
                    positionOffsetPixels: Int
                ) {

                }

                override fun onPageSelected(position: Int) {
                    save_position_azkar = position;

                }

                override fun onPageScrollStateChanged(state: Int) {
                }

            });
            binding.SwipePagesActivityVP.setAdapter(adapterForAzkarSwipe);
            if (SharedPerefrenceHelper.getBooleanForAzkar(this, IS_TRUE_AZKAR, false)) {
                ShowDialog.showDialogForRetrieveAzkar(
                    this,
                    binding.SwipePagesActivityVP,
                    position_azkar
                );
            } else {
                binding.SwipePagesActivityVP.setCurrentItem(position_azkar);
            }
        }
    }

    private fun declareAdapter() {
        val adapterForSwipe = AdapterForSwipe(
            images!!,object :ClickListener<Int>{
                override fun onClick(view: View?, position: Int) {
                    DialogUtils.showDialog(this@SwipePagesActivity,getString(R.string.save_position),object:View.OnClickListener{
                        override fun onClick(p0: View?) {
                            dataStoreViewModel.deleteReadingQuran()
                            dataStoreViewModel.saveReadingQuran(save_position)
                            finish()
                        }
                    })
                }
            }
        )
        binding.SwipePagesActivityVP.adapter = adapterForSwipe
    }

    private fun observeViewPager() {
        binding.SwipePagesActivityVP.addOnPageChangeListener(object :
            ViewPager.OnPageChangeListener {
            override fun onPageScrolled(
                position: Int,
                positionOffset: Float,
                positionOffsetPixels: Int
            ) {
            }

            override fun onPageSelected(position: Int) {
                save_position = position
            }

            override fun onPageScrollStateChanged(state: Int) {
            }

        })
    }

    private fun observeSaveData() {
        dataStoreViewModel.getReadingQuran.asLiveData().observe(this){
            if (it >=0 ){
                DialogUtils.showDialogForRetrieveReading(this,getString(R.string.app_name),getString(R.string.do_want_Save),
                    { _, _ ->
                        binding.SwipePagesActivityVP.currentItem = it
                    }
                ) { _, _ ->
                    dataStoreViewModel.deleteReadingQuran()
                    binding.SwipePagesActivityVP.currentItem = position
                }
            }else
                binding.SwipePagesActivityVP.currentItem = position
        }

    }

    private fun getScreenOn(isScreenOn: Boolean) {
        if (isScreenOn) {
            window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
        } else {
            window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
        }
    }


    private fun createImage() {
        for (i in images!!.indices) {
            bundle!!.putInt(SwarFragment.SAVE_IMAGES, images!![i])

            bundle!!.putInt(SAVE_POSITION, position)
        }
        binding.SwipePagesActivityPB.visibility = View.GONE
    }

    private val argemnetsNotification: Unit
        private get() {
            if (bundle != null) {
                imagesNotification =
                    bundle!!.getIntegerArrayList(AlarmReceiver.SAVE_Position_Notification)
                Log.d("TAG", " $imagesNotification")
            }
        }

    private fun createImageNotification() {
        if (imagesNotification != null) {
            for (i in imagesNotification!!.indices) {
                bundle!!.putInt(AlarmReceiver.SAVE_Position_Notification, imagesNotification!![i])
            }
            Log.d("TAG", " $imagesNotification")
            binding.SwipePagesActivityPB.visibility = View.GONE
        }
    }

    private val azkarNotification: Unit
        private get() {
            if (bundle != null) {
                imagesNotification =
                    bundle!!.getIntegerArrayList(AlarmReceiver.SAVE_Position_Notification)
                Log.d("TAG", " $imagesNotification")
            }
        }

    private fun createAzkarNotification() {
        if (imagesNotification != null) {
            for (i in imagesNotification!!.indices) {
                bundle!!.putInt(AlarmReceiver.SAVE_Position_Notification, imagesNotification!![i])
            }
            Log.d("TAG", " $imagesNotification")
            binding.SwipePagesActivityPB.visibility = View.GONE
        }
    }

    private fun getImagesFirst() {
        if (bundle != null) {
            imagesFirst = bundle!!.getIntegerArrayList(NavigationDrawaberActivity.SAVE_ALL_IMAGES)
        }
    }

    private fun createImagesFirst() {
        if (imagesFirst != null) {
            for (i in imagesFirst!!.indices) {
                bundle!!.putInt(NavigationDrawaberActivity.SAVE_ALL_IMAGES, imagesFirst!![i])
            }
            binding.SwipePagesActivityPB.visibility = View.GONE
        }
    }

    // modelAzkarList =  Gson().fromJson(st, listType);
    fun getAzkarList() {
        if (bundle != null) {
            val listType = object : TypeToken<List<ModelAzkar?>?>() {}.type
            val st = bundle!!.getString(AzkarFragment.SAVE_AZKAR)
            modelAzkarList = Gson().fromJson(st, listType);
            position_azkar = bundle!!.getInt(AzkarFragment.SAVE_POTION_AZKAR)
        }
        binding.SwipePagesActivityPB.visibility = View.GONE
    }

    //    modelAzkarList = Gson().fromJson(st, listType);
    private val argemnetsForNotificationAzkar: Unit
        private get() {
            if (bundle != null) {
                val listType = object : TypeToken<List<ModelAzkar?>?>() {}.type
                val st = bundle!!.getString(MorningAzkarAlarmReceiver.SAVE_POSITION_MORNING_AZKAR)
                modelAzkarList = Gson().fromJson(st, listType);
                position_azkar =
                    bundle!!.getInt(MorningAzkarAlarmReceiver.NOTIFICATION_ID_NUMBER_AZKAR)
            }
            binding.SwipePagesActivityPB.visibility = View.GONE
        }

    override fun onSaveInstanceState(outState: Bundle) {
        SharedPerefrenceHelper.putInt(this, SwarFragment.SAVE_IMAGES, save_position)
        SharedPerefrenceHelper.putInt(this, AzkarFragment.SAVE_AZKAR, save_position_azkar)
        super.onSaveInstanceState(outState)
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        save_position = savedInstanceState.getInt(SwarFragment.SAVE_IMAGES)
        save_position_azkar = savedInstanceState.getInt(AzkarFragment.SAVE_AZKAR)
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onBackPressed() {
        super.onBackPressed()
        overridePendingTransition(R.anim.item_anim_no_thing, R.anim.item_anim_slide_from_bottom)
    }

    companion object {
        const val IS_TRUE = "is_true"
        const val IS_TRUE_AZKAR = "is_true_azkar"
    }
}